src/components/media-player/media-player-track.js

```diff
@@ -30,6 +31,7 @@ function MediaPlayerTrack({
   const playerRef = useRef();
   const playerHandlersRef = useRef({});
   const [sourceDuration, setSourceDuration] = useState(0);
+  const youtubeThumbnailUrl = useYoutubeThumbnailUrl(sourceUrl);
```

Comment:
Maybe it's better to `analyzeMediaUrl(sourceUrl)` directly inside `useYoutubeThumbnailUrl` and do an early return of `null` in case the source is not Youtube.

I find it confusing to always have a `youtubeThumbnailUrl` value even though the source could be internal.

Reply:
True. I left the check in the `youtube-thumbnail-cache` though, rather then putting it into the hook, but it returns `null` for media other than Youtube.